Framework/Algorithms/src/MaxEnt.cpp

```diff
@@ -350,19 +420,42 @@ void MaxEnt::exec() {
   iterationCounts.reserve(nSpec);
   outEvolChi->setPoints(0, Points(nIter, LinearGenerator(0.0, 1.0)));
 
+  size_t dataLength = complexData ? 2 * inWS->y(0).size() : inWS->y(0).size();
+
   for (size_t spec = 0; spec < nSpec; spec++) {
 
     // Start distribution (flat background)
     std::vector<double> image(npoints, background);
 
-    std::vector<double> data;
-    std::vector<double> errors;
+    std::vector<double> data(dataLength, 0.0);
+    std::vector<double> errors(dataLength, 0.0);
     if (complexData) {
-      data = toComplex(inWS, spec, false);  // false -> data
-      errors = toComplex(inWS, spec, true); // true -> errors
+      data = toComplex(inWS, spec, false, sumSpectra);  // false -> data
```

Comment:
I don't understand these comments. Could you please elaborate?

Reply:
It determines whether the complex data is taken from the errors or the Y values. The comments could be improved. That was how they were before I added the 4th argument.